sap/src/main/java/nl/nn/adapterframework/extensions/sap/jco3/SapListenerImpl.java

```diff
@@ -149,36 +151,62 @@ public String getPhysicalDestinationName() {
 	}
 
 	@Override
-	public RawMessageWrapper<JCoFunction> wrapRawMessage(JCoFunction jcoFunction, PipeLineSession session) throws ListenerException {
-		return new RawMessageWrapper<>(jcoFunction, getCorrelationIdFromField(jcoFunction), null);
+	public RawMessageWrapper<M> wrapRawMessage(M jcoFunction, PipeLineSession session) throws ListenerException {
+		if(jcoFunction instanceof JCoFunction) {
```

Comment:
Kan je hier pattern matching instanceof gebruiken? 

```suggestion
		if (jcoFunction instanceof JCoFunction function) {
```

Wellicht wil je dan nog `jcoFunction` renamen om het leesbaarder te maken.

Reply:
Java 8 gaat dat niet leuk vinden.

Reply:
ah, ik wist niet dat dit nog op 8 draaide .. :-/

Reply:
Deze fix is voor 7.9 en dat is de laatste release waar we nog Java8 moeten ondersteunen.

De forward-port naar 8.3/9.0 kan wel deze pattern matching gebruiken.